enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToSlave.java

```diff
@@ -262,8 +263,9 @@ public URI switchToSlave( LifeSupport haCommunicationLife, URI me, URI masterUri
 
         monitor.switchToSlaveStarted();
 
-        // Wait for current transactions to stop first
-        long deadline = SYSTEM_CLOCK.currentTimeMillis() + config.get( HaSettings.state_switch_timeout );
+        // Wait a short while for current transactions to stop first, just to be nice.
+        // We can't wait forever since switching to our designated role is quite important.
+        long deadline = SYSTEM_CLOCK.currentTimeMillis() + config.get( HaSettings.internal_state_switch_timeout );
```

Comment:
True, that sounds like a reasonable thing to do. 10 seconds is an just an arbitrary amount of seconds, although certainly less than 120 which the case before this change. And since we've concluded that it's only read transactions that we are waiting for then 10 seconds is a pretty long time.